namespaces/resource-namespace.md

```diff
@@ -0,0 +1,328 @@
+
+### Data Structure (MSON Element)
+
+Data structure definition using MSON namespace elements.
+
+#### Properties
+
+- `element`: dataStructure (string, fixed)
+
+### Asset (Element)
+
+Arbitrary data asset.
+
+#### Properties
+
+- `element`: asset (string, fixed)
+- `meta`
+    - `class` (array, fixed)
+        - (enum[string])
+            - messageBody - Asset is an example of message-body
+            - messageBodySchema - Asset is an schema for message-body
+- `attributes` (object)
+    - `contentType` (string) - Optional media type of the asset
+    - `href` (Href) - Link to the asset
+- `content` (string) - A textual representation of the asset
+
+## Resource Elements
+
+### Resource (Element)
+
+The Resource representation with its available transitions and attributes.
+
+#### Properties
+
+- `element`: resource (string, fixed)
+- `attributes` (object)
+    - `href` (Templated Href) - URI Template of this resource.
+    - `hrefVariables` (Href Variables) - Definition of URI Template variables used in the `href` property.
+- `content` (array)
+    - (Transition) - State transition available for this resource.
+
+        The `content` MAY include multiple `Transition` elements.
+
+    - (Data Structure) - Data structure representing the resource.
+
+        The `content` MUST NOT include more than one `Data Structure`.
+
+#### Example
+
+```json
+{
+    "element": "resource",
+    "meta": {
+        "title": "Question",
+        "description": "A Question object has the following attributes."
+    },
+    "attributes": {
+        "href": "/questions/{question_id}",
+        "hrefVariables": {
+            "element": "hrefVariables",
+            "attributes": {
+                "name": "question_id"
+            },
+            "content": null
+        }
+    },
+    "content": [
+        {
+            "element": "dataStructure"
+        }
+    ]
+}
+```
+
+### Transition (Element)
+
+A transition is an available progression from one state to another state.
+Exercising a transition initiates a transaction.
+
+The content of this element is array of protocol-specific transactions.
+
+Note: At the moment only the HTTP protocol is supported.
+
+#### Properties
+
+- `element`: transition (string, fixed)
+- `attributes` (object)
+    - `relation` - (string) - Link relation type as specified in [RFC 5988][].
+
+        The value of `relation` attribute SHOULD be interpreted as a link relation
+        between transition's parent resource and the transition's target resource
+        as specified in the `href` attribute.
+
+    - `href` (Templated Href) - The URI template for this transition.
+
+        If present, the value of the `href` attribute SHOULD be used to resolve
+        target URI of the transition.
+
+        If not set, the parent `resource` element `href` attribute SHOULD be
+        used to resolve the target URI of the transition.
+
+    - `parameters` (Href Variables) - Input parameters.
```

Comment:
Semantically, the two are different.

Let's start with transition: Any transition may have input properties (refer to representor charter) those properties are either parameters or attributes. In addition to that – since a templated href can be provided – you can have some other variables in there. 

Consider resource "friends" which is an array of friend resources every user have. Its templated href may be something like `/user/{username}/friends`. Now if you are "sitting" on the `Friends` resource you can perhaps take a transition `list` that lists all the friends. Let's say the transition parameter is `limit` to limit maximum number of friend resources returned. The full templated href would look like `/user/{username}/friends{?limit}`. 

You can see that `username` and `limit` are input parameters of the transition. In a resource context, those are not necessarily parameters. 

Now, strictly speaking, the parameter of the `list` transition is really just a `limit` (not `username`) because in a hypermedia system the value of `username` should be already resolved by now. 

So in theory the transition should have both `hrefVariables` and `parameters`. I didn't want to overcomplicate things introducing both here but I've wanted to at least acknowledge this fact.

Does this make sense?